### PR TITLE
Fix visible endpoints missing issues in the function block

### PR DESCRIPTION
**Pull request by kanushka**
## Purpose

- Fix visible endpoints missing issues in the function block.
- Add visible endpoints to block-level statements.
- Add line range filter to visible endpoint list.
- Remove duplications in the visible endpoint list.

## Approach
Update visible endpoint object with endpoint declaration position.
Then `transformSyntaxNode` visitor has updated to capture block statements and filter-out local (inner) endpoints.
And add visible endpoints to every block statement. 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by NipunaMarcus — APPROVED** [review]

File: misc/diagram-util/src/main/java/org/ballerinalang/diagramutil/SyntaxTreeMapGenerator.java [hunk 1]
```diff
@@ -200,20 +197,57 @@ protected JsonElement transformSyntaxNode(Node node) {
...
+                        if (epStartLine >= node.lineRange().startLine().line()
+                                && epEndLine < node.lineRange().endLine().line()
+                                && !endpoint.has("innerBlock")) {
+                            blockEndpoints.add(endpoint);
+                            // Add key to filter endpoints only visible to a block
+                            endpoint.addProperty("innerBlock", true);
+                        }
+                    }
+                }
+                nodeJson.add("VisibleEndpoints", blockEndpoints);
```
Comment (NipunaMarcus) [inline misc/diagram-util/src/main/java/org/ballerinalang/diagramutil/SyntaxTreeMapGenerator.java:230]: won't this logic fails to rightfully detect what shown to the block level ? coz I didn't see a line where we remove endpoints that already been classified as belong inside the block ... So if we have a code like below
```
function name() {
  <endpoint1>
   if true {
      <endpoint2>
  }
}
```
as visible endpoint classification happens at the end won't `endpoint2` will be added to the visible list of function block as it is still in the visible endpoint list and it is between function block start and end?

Comment (kanushka) [reply misc/diagram-util/src/main/java/org/ballerinalang/diagramutil/SyntaxTreeMapGenerator.java:230]: This block will add all block-level endpoints and mark them as an `inner block` endpoint. https://github.com/ballerina-platform/ballerina-lang/blob/257bef32df4f35bf2c1c48eacb40fbdc574d5ff6/misc/diagram-util/src/main/java/org/ballerinalang/diagramutil/SyntaxTreeMapGenerator.java#L221-L227

And all other endpoints above the selected block and inside the function block will detect by the below condition. `getParentFunctionBlock` function will recursively get the function body node to detect the function body scope. https://github.com/ballerina-platform/ballerina-lang/blob/257bef32df4f35bf2c1c48eacb40fbdc574d5ff6/misc/diagram-util/src/main/java/org/ballerinalang/diagramutil/SyntaxTreeMapGenerator.java#L213-L219 
I have validated inner level endpoints using the `interlock` field.

Comment (NipunaMarcus) [reply misc/diagram-util/src/main/java/org/ballerinalang/diagramutil/SyntaxTreeMapGenerator.java:230]: ack

**Comment by kanushka** [comment]
We have merged duplicate PR https://github.com/ballerina-platform/ballerina-lang/pull/35690 with [2201.0.3-stage-new](https://github.com/ballerina-platform/ballerina-lang/tree/2201.0.3-stage-new) branch. 
I'm closing this PR since this temporary branch will sync with the [2201.0.x](https://github.com/ballerina-platform/ballerina-lang/tree/2201.0.x) branch.

**Pull request closed without merging**